src/test_pages/BasePage.ts

```diff
@@ -0,0 +1,33 @@
+import { Locator, Page, expect, test } from '@playwright/test';
+
+export abstract class BasePage{
+    constructor(protected page: Page){
+    }
+
+    protected async validatePageUrl(url: string){
```

Comment:
why not `page.waitForUrl`?

Reply:
i would recommend you to use `Prettier - Code Formatter` extension. 
![image](https://github.com/hasadna/open-bus-map-search/assets/1336862/6f0f3971-e8bd-4a92-aaf3-65a63baed08a)
after installing it  - you can press ALT + SHIFT + F and all lint errors automatically get fixed

Reply:
> i would recommend you to use `Prettier - Code Formatter` extension. ![image](https://private-user-images.githubusercontent.com/1336862/292774228-6f0f3971-e8bd-4a92-aaf3-65a63baed08a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM1MzcxODIsIm5iZiI6MTcwMzUzNjg4MiwicGF0aCI6Ii8xMzM2ODYyLzI5Mjc3NDIyOC02ZjBmMzk3MS1lOGJkLTRhOTItYWFmMy02NWE2M2JhZWQwOGEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMjVUMjA0MTIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTgyNmUxNTBiN2I1NmViZGEzZmM5NzQ0ZDY4ZWYzYTljNGNiZmJjZTdmNmFjYzdjODc2OWVlYzM3YWYxYzc0NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.TXROP2d5nvhRWzRM0J2I0UlhO70Twi9ECdXmxTWSpWA) after installing it - you can press ALT + SHIFT + F and all lint errors automatically get fixed

Great advise! Thanks

Reply:
> advise

Actually both options will give the same result. Because both are asynchronous and both have the option to set the timeout, which makes it the equivalent of expect(this.page).toHaveURL(url). But I like to use the expect because it's used as an assertion within the test case, providing clearer test output. If for example the assertion fails, it explicitly states that the URL doesn't match the expected value, making debugging easier.

Reply:
it's not the same. `waitForUrl` will wait for navigation to finish

Reply:
expect(this.page).toHaveURL(url) inside of it will wait untill the navigation will finish and will behave exactly the same. I use it in my testcases and it works great.

Reply:
good to know.
anyway, won't it be easier to place the expect statement in the original test?